lib/benchee/formatters/html.ex

```diff
@@ -80,23 +80,29 @@ defmodule Benchee.Formatters.HTML do
     particular job (one per benchmark input)
   """
   @spec output(Suite.t) :: Suite.t
-  def output(suite = %{configuration: %{formatter_options: %{html: %{file: _}}}}) do
+  def output(suite = %{configuration: %{formatter_options: %{html: %{file: _, auto_open: auto_open?}}}}) do
     suite
     |> format
     |> write
 
-    open_report(suite)
+    if auto_open?, do: open_report(suite)
+    suite
   end
 
   @default_filename "benchmark_output/my.html"
+  @default_auto_open true
   def output(suite) do
     suite
-    |> update_filename()
+    |> fallback_default_config()
     |> output()
   end
 
-  defp update_filename(suite) do
-    updated_configuration = %Configuration{suite.configuration | formatter_options: %{html: %{file: @default_filename}}}
+  defp fallback_default_config(suite) do
+    opts = suite.configuration.formatter_options
+           |> Map.get(:html, %{})
+           |> Map.put_new(:file, @default_filename)
+           |> Map.put_new(:auto_open, @default_auto_open) 
```

Comment:
huh, nice usage of `put_new` haven't used it yet for such a behaviour, but I should :)